tests/unit/s3middleware/tagging.js

```diff
@@ -0,0 +1,21 @@
+const assert = require('assert');
+const { _validator } = require('../../../lib/s3middleware/tagging');
+
+describe('tagging validator', () => {
+    it('validates keys and values are less than 128 and 256', () => {
+        const result = _validator.validateKeyValue('hey', 'you guys');
+        assert.strictEqual(result, true);
+    });
+    it('returns error for keys greater than 128', () => {
+        const result = _validator.validateKeyValue('Y'.repeat(200), 'you guys');
+        assert(result instanceof Error);
```

Comment:
There is `assert.throws()` for this case.

Reply:
the validation doesn't actually throw an error though it just returns one. ( I commented on this in slack about it ) It would fail CI/CD if it threw one.

Reply:
I didn't know if this API was already widely used or not. If it's already used everywhere, no point in changing, but otherwise it's a suboptimal API to just return errors as if the call were asynchronous while it's not.

Reply:
I agree but it felt a bit out of scope for my task, it also sounds like a backward-incompatible change we would have to fix whatever does use it. And at the very least the method is being used in S3C when putting new tags on objects.